drivers/at/at.c

```diff
@@ -88,18 +96,54 @@ static void _isrpipe_write_one_wrapper(void *_dev, uint8_t data)
 #endif
 }
 
-int at_dev_init(at_dev_t *dev, uart_t uart, uint32_t baudrate, char *buf, size_t bufsize)
+int at_dev_init(at_dev_t *dev, at_dev_init_t const *init)
 {
-    dev->uart = uart;
+    dev->uart = init->uart;
+    dev->err_buf = init->err_buf;
+    dev->err_buf_size = init->err_buf_size;
 
 #if IS_USED(MODULE_AT_URC_ISR)
     dev->awaiting_response = false;
     dev->event.handler = _event_process_urc;
 #endif
 
-    isrpipe_init(&dev->isrpipe, (uint8_t *)buf, bufsize);
+    isrpipe_init(&dev->isrpipe, (uint8_t *)init->rx_buf, init->rx_buf_size);
 
-    return uart_init(uart, baudrate, _isrpipe_write_one_wrapper, dev);
+    return uart_init(init->uart, init->baudrate, _isrpipe_write_one_wrapper, dev);
+}
+/**
+ * @retval  0, if the response is "OK"
+ * @retval -AT_ERR_EXTENDED, if the response in "+CMx ERROR: <err>", and <err>
+ *          has been successfully copied to @p dev->err_buf
+ * @retval -1, if the response is "ERROR", or "+CMx ERROR: <err>" but <err>
+ *          could not be copied
+ * @retval  1 otherwise
+ */
+static int parse_resp(at_dev_t *dev, char const *resp)
+{
+    if (*resp == '\0') {
+        return 1;
+    }
+    if (strncmp(CONFIG_AT_RECV_OK, resp, strlen(CONFIG_AT_RECV_OK)) == 0) {
```

Comment:
Why do we use `strncmp` here? The only reason see is if the input (`resp`) is not a NUL-terminated string. Is that the case?

I'm guessing that `strcmp(CONFIG_AT_RECV_OK, resp) == 0` is even faster than `strncmp`.

Reply:
The string passed is always 0-terminated. However, in `at_send_cmd_get_lines()`, the answer might be `OK\r`, depending on whether `keep_eol` is true or not.

Reply:
Ah OK